asv_bench/benchmarks/merge.py

```diff
@@ -13,6 +15,45 @@ def setup(self, existing_elements):
             d[f"var{i}"] = i
         self.dataset = xr.merge([d])
 
-    def time_variable_insertion(self, existin_elements):
+        d = {"set_2_{i}": i for i in range(existing_elements)}
+        self.dataset2 = xr.merge([d])
+
+    def time_variable_insertion(self, existing_elements):
         dataset = self.dataset
         dataset["new_var"] = 0
+
+    def time_merge_two_datasets(self, existing_elements):
+        xr.merge([self.dataset, self.dataset2])
+
+    def time_dict_of_dataarrays_to_dataset(self, existing_elements):
+        # The idea here is to time how long it takes to go from numpy
+        # and python data types, to a full dataset
+        # See discussion
+        # https://github.com/pydata/xarray/issues/7224#issuecomment-1292216344
+        value = np.array(["0", "b"], dtype=str)
+        time = np.array([0, 1])
+        coords = dict(time=time)
```

Comment:
Should this be created in setup? For 1000 elements it probably doesn't matter but for 0 or 1 it might.

Reply:
ok, i can move this.